From 0539613784783f1dfa1bc4f87315196a1d76f11c Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 08 Apr 2019 18:16:24 +0300
Subject: [PATCH] [SW-0] spi: BUG - using not initialized BSS variable

In preboot we do not clear BSS so using non initialized static variables
is wrong.

Change-Id: I3770b1f1bf732b12fef1b3f8011bdfc534c304e8
---

diff --git a/common/spi.c b/common/spi.c
index 532d8a8..bdaed34 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -54,22 +54,12 @@
 
 static void spi_claim_bus(void)
 {
-	static u64 reg;
-
-	if (!reg)
-		reg = get_cs_pad_sel();
-
-	regmap_writel(1, reg);
+	regmap_writel(1, get_cs_pad_sel());
 }
 
 static void spi_release_bus(void)
 {
-	static u64 reg;
-
-	if (!reg)
-		reg = get_cs_pad_sel();
-
-	regmap_writel(0, reg);
+	regmap_writel(0, get_cs_pad_sel());
 }
 
 static inline void spi_dev_update_frame_size(unsigned int len)
